app/models/activity.rb

```diff
@@ -157,4 +157,12 @@ def transactions_total
   def planned_disbursements_total
     planned_disbursements.sum(:value)
   end
+
+  def actual_total_for_report(report:)
+    transactions.where(report: report, date: report.created_at.all_quarter).sum(:value)
```

Comment:
Is `created_at` the right date to use here? `financial_year` and `financial_quarter` are both created from the date the Report is created, so I assume it's alright to use the `created_at` date to also determine the quarter range?

Reply:
We've just chatted about this but I think you are right, as the `financial_year` and `financial_quarter` are both derived from the date the Report is created, calculating the date range can be done like this.

Reply:
Why is the date filter necessary? I thought we were associating transactions to reports to make that relationship explicit, rather than inferring it from dates. It feels like a brittle assumption to say that a report relates to the quarter during which it was created in the database.

Reply:
Feels like I am still failing to communicate all this so apologies! :( 

The stance we are taking is that there is only ever one report for a given org and fund. The report is seen as 'the data for a quarter' but I believe it is important to allow transactions that are not in the quarter to be reported at the same time, an example of this would be a correction to a previous reported transaction.

Whilst we know little about how this 'correction' will take place, I think its fair to assume it will be reported via the same mechanism. For this reason, in this context, for transactions, when reviewing, the users will want to see the total transaction value per activity for the quarter in question and this should not include any values that fall outside of the quater date range as this would make the figures inaccurate.

We currently have no validation to ensure the transactions added during a report are only from the quarter in question so this scoping guards against that issue. Once we know more about making corrections we may be able to remove the scope and be confident that transactions associated to a report are only for the date range users care about during the review.

Looking forward we may want to handle corrections in some other way - who knows but for now I feel we need this guard in place.

I agree that the relationship between the creation date and the quarter feeling brittle, but I am confident that as the application is in full control of the creation of the reports (users cannot create them) we can guard against this also and we are mindful of this need going into the work to create the next report.

Sorry for the huge comment - happy (and would rather) talk it though if things are not clear.